src/nvim/testdir/test_assert.vim

```diff
@@ -0,0 +1,12 @@
+" Test that the methods used for testing work.
+
+func Test_assert_notequal()
```

Comment:
this should be added to `test/functional/legacy/assert_spec.lua` instead.

Also it won't run as it is, because it isn't in `src/nvim/testdir/Makefile`